### ISSUE TITLE
Computing class inheritance relations → segmentation fault when run against vulkan-hpp headers

### ISSUE THREAD

**Issue by triffid** [doxygen/doxygen#10951]
**Describe the bug**
Doxygen segfaults when run against the contents of https://github.com/KhronosGroup/Vulkan-Hpp/tree/main/vulkan/ at "Computing class inheritance relations…"

**To Reproduce**
1) clone https://github.com/KhronosGroup/Vulkan-Hpp
2) cd Vulkan-Hpp/vulkan
3) generate a basic doxyfile for C++ - I just used doxywizard, changed basically nothing, and hit save
4) run doxygen

**Expected behavior**
Generates html describing the headers, or at least prints a normal error if there's a problem

**Version**
doxygen 1.9.8

Gentoo Linux x86_64 (doxygen compiled locally using Gentoo's ebuild)

**Stack trace**
```
Doxygen version used: 1.9.8
Searching for include files...
Searching for example files...
Searching for images...
Searching for dot files...
Searching for msc files...
Searching for dia files...
Searching for files to exclude
Searching INPUT for files to process...
Searching for files in directory /home/triffid/Projects/Vulkan-Hpp/vulkan
Reading and parsing tag files
Parsing files
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan.cppm...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan.cppm...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_enums.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_enums.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_extension_inspection.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_extension_inspection.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_format_traits.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_format_traits.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_funcs.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_funcs.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_handles.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_handles.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_hash.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_hash.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_hpp_macros.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_hpp_macros.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_raii.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_raii.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_shared.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_shared.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_static_assertions.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_static_assertions.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_structs.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_structs.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_to_string.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_to_string.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_video.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkan_video.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc.cppm...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc.cppm...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_enums.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_enums.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_extension_inspection.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_extension_inspection.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_format_traits.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_format_traits.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_funcs.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_funcs.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_handles.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_handles.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_hash.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_hash.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_hpp_macros.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_hpp_macros.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_raii.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_raii.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_shared.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_shared.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_static_assertions.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_static_assertions.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_structs.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_structs.hpp...
Preprocessing /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_to_string.hpp...
Parsing file /home/triffid/Projects/Vulkan-Hpp/vulkan/vulkansc_to_string.hpp...
Building macro definition list...
Building group list...
Building directory list...
Building namespace list...
Building file list...
Building class list...
Building concept list...
Computing nesting relations for classes...
Associating documentation with classes...
Associating documentation with concepts...
Associating documentation with modules...
Building example list...
Searching for enumerations...
Searching for documented typedefs...
Searching for members imported via using declarations...
Searching for included using directives...
Searching for documented variables...
Building interface member list...
Building member list...
Searching for friends...
Searching for documented defines...
Computing class inheritance relations...

Program received signal SIGSEGV, Segmentation fault.
0x000055555598cffe in SymbolResolver::Private::substTypedef(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, Definition const*, QCString const&, MemberDef const**) ()
(gdb) bt
#0  0x000055555598cffe in SymbolResolver::Private::substTypedef(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, Definition const*, QCString const&, MemberDef const**) ()
#1  0x0000555555989199 in SymbolResolver::Private::followPath(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, Definition const*, QCString const&) ()
#2  0x000055555598abf0 in SymbolResolver::Private::isAccessibleFromWithExpScope(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Definition const*, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Definition const*> > >&, AccessStack&, Definition const*, Definition const*, QCString const&) ()
#3  0x000055555598b07f in SymbolResolver::Private::isAccessibleFromWithExpScope(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Definition const*, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Definition const*> > >&, AccessStack&, Definition const*, Definition const*, QCString const&) ()
#4  0x000055555598d874 in SymbolResolver::Private::getResolvedType(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, Definition const*, Definition const*, QCString const&, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, int&, ClassDef const*&, MemberDef const*&, QCString&, QCString&) ()
#5  0x0000555555987f3b in SymbolResolver::Private::getResolvedTypeRec(std::unordered_set<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, Definition const*, QCString const&, MemberDef const**, QCString*, QCString*) ()
#6  0x000055555598def8 in SymbolResolver::resolveClass(Definition const*, QCString const&, bool, bool) ()
#7  0x00005555555fcabf in findClassWithinClassContext(Definition*, ClassDef*, QCString const&) ()
#8  0x000055555564299d in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#9  0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#10 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#12 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#13 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#14 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#15 0x000055555564588d in findBaseClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, FindBaseClassRelation_Mode, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#16 0x0000555555644023 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#17 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#18 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#19 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#20 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#21 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#22 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#23 0x000055555564588d in findBaseClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, FindBaseClassRelation_Mode, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
```
and so on for over 11,000 frames, but just seems to be recursing through the same two functions the whole way:
<details>

```
… «11350 essentially identical frames skipped»
#11374 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11375 0x000055555564588d in findBaseClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, FindBaseClassRelation_Mode, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11376 0x0000555555644023 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11377 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11378 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11379 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11380 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11381 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11382 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11383 0x000055555564588d in findBaseClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, FindBaseClassRelation_Mode, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11384 0x0000555555644023 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11385 0x0000555555646baf in findUsedClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11386 0x0000555555644052 in findClassRelation(Entry const*, Definition*, ClassDefMutable*, BaseInfo const*, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&, FindBaseClassRelation_Mode, bool) ()
#11387 0x000055555564588d in findBaseClassesForClass(Entry const*, Definition*, ClassDefMutable*, ClassDefMutable*, FindBaseClassRelation_Mode, bool, std::unique_ptr<ArgumentList, std::default_delete<ArgumentList> > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, int> > > const&) ()
#11388 0x0000555555645dd3 in findInheritedTemplateInstances() ()
#11389 0x00005555556610e1 in parseInput() ()
#11390 0x00005555555db44d in main ()
```
</details>

**Additional context**
I just want some proper documentation for Vulkan-hpp's class hierarchy; apparently none exists anywhere - available documentation basically just says "we basically copied the C library and changed stuff around, good luck"

**Comment by albert-github** [comment]
I did a quick test with the current master version of doxygen under Cygwin (1.12.0 (b3c28ea6cc7ee2f7ece9c3f7539ca01cc8b36e5f)) and got also a crash with an default Doxyfile.

As the provided example (master version of Vulkan-Hpp) is not persistent I added the used code as example:
Example: [example.tar.gz](https://github.com/user-attachments/files/15899627/example.tar.gz)

**Comment by albert-github** [comment]
I reduced the problem significantly to: [example1.tar.gz](https://github.com/user-attachments/files/16017721/example1.tar.gz)

The place of the crash looks like a different place than the initial mentioned place, so maybe there are even multiple problems ( :-( )

**Comment by doxygen** [comment]
@triffid Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
